### PR TITLE
Fix minor bugs and code refactoring

### PR DESCRIPTION
**Pull request by ssuminahn**
:Release Notes:
Fix minor bugs and code refactoring

:Detailed Notes:
- Convert function that generates date format file name to a library
- Modify repeated code into a common function
- Fix the part where the config created by the command is different from the git file
- Sort order of source require

:Testing Performed:
1. Pass unit test on ose
2. Pass eslint
3. Check below commands and results
3-1. ares-config -p ose
    > check git status and no diff files/config/config.json
3-2. Check each operation of ares-install/launch/device

:Issues Addressed:
[WRN-1725] Fix minor bugs and code refactoring of CLI

### CODE REVIEW

**Review by ssuminahn — COMMENTED** [review]
code :+1: 
verify :+1:

File: bin/ares-setup-device.js [hunk 1]
```diff
@@ -102,6 +101,7 @@ if (argv.list) {
 } else if (argv.search || argv.timeout) {
     op = search;
 } else if (argv.add || argv.modify || argv.info) {
+    // console.log("?")
```
Comment (YoungeunKim) [inline bin/ares-setup-device.js:104]: Please remove this comment.

File: lib/device.js [hunk 2]
```diff
@@ -285,7 +267,7 @@ const util = require('util'),
                     captureFormat = "PNG"; // PNG is default format
 
                 if (options.outputPath === null){
-                    captureFileName = _createDateFileName();
+                    captureFileName = options.session.target.name + "_" + "display" + options.display + "_" + createDateFileName("") + ".png";
```
Comment (YoungeunKim) [inline lib/device.js:270]: Why don't you modify this? 
createDateFileName("", ".png")? 
And make sure what is the first parameter or add Null? 
Let's have a talk.

File: lib/install.js [hunk 3]
```diff
@@ -464,6 +448,15 @@ const fs = require('fs'),
         }
     };
 
+    function makeSession(options, next){
+        log.info("install#makeSession");
+        if (!options.session) {
+            options.session = new novacom.Session(options.device, next);
+        } else {
+            next(null, options.session);
```
Comment (YoungeunKim) [inline lib/install.js:456]: Need to check ares-install opkg command.
Cause 375 line is "function(next)"

File: lib/install.js [hunk 4]
```diff
@@ -381,12 +370,7 @@ const fs = require('fs'),
             async.series([
                 function(next) {
                     options.nReplies = 1; // -n 1
-                    if (!options.session) {
-                        const printTarget = true;
-                        options.session = new novacom.Session(options.device, printTarget, next);
```
Comment (YoungeunKim) [inline lib/install.js:386]: how to handle "printTarget" parameter in your change??

File: lib/log.js [hunk 5]
```diff
@@ -339,7 +340,7 @@ const util = require('util'),
     function createLogFile(options, next) {
         log.info("log#createLogFile");
         if (options.argv.argv.remain.length === 0) {
-            options.argv.argv.remain[0] = path.resolve(_createDefaultFileName());
+            options.argv.argv.remain[0] = path.resolve(createDateFileName("_")+".log");
```
Comment (YoungeunKim) [inline lib/log.js:343]: Need to understand the parameter. 
Why don't you use such as "formatType"?

File: lib/session.js [hunk 6]
```diff
@@ -5,8 +5,8 @@
  */
 
 const npmlog = require('npmlog'),
-    luna = require('./base/luna'),
-    errHndl = require('./base/error-handler');
+errHndl = require('./base/error-handler'),
```
Comment (YoungeunKim) [inline lib/session.js:8]: Please add 4 spaces.

**Review by YoungeunKim — CHANGES_REQUESTED** [review]

**Comment by YoungeunKim** [comment]
Could you update npm-shrinkwrap.json? The reason is that "ora" is added package.json in "add spinner" commit.

File: bin/ares-install.js [hunk 7]
```diff
@@ -6,24 +6,23 @@
...
-    log = require('npmlog'),
+const async = require('async'),
+    fs = require('fs'),
     nopt = require('nopt'),
+    log = require('npmlog'),
+    path = require('path'),
     installLib = require('./../lib/install'),
     commonTools = require('./../lib/base/common-tools'),
-    convertJsonToList = require('./../lib/util/json').convertJsonToList,
-    spinner = require('../lib/util/spinner');
```
Comment (YoungeunKim) [inline bin/ares-install.js:17]: spinner have been removed in your change. Please add it.

**Comment by YoungeunKim** [comment]
@ssuminahn < 
All unit test passed? When I ran unit test, it failed because "spinner" importing has been removed. Please check and verify it again. 

Code :-1: 
Verified :-1:

File: lib/install.js (hunk 3 above)
Comment (ssuminahn) [reply lib/install.js:458]: I've checked opkg insttall and that is working fine.

**Review by ssuminahn — COMMENTED** [review]
code :+1: 
verify :+1:

File: lib/util/createFileName.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+        const curDate = new Date();
+        const dateFormat = __pad(curDate.getFullYear(), 2)
+            +  __pad((curDate.getMonth()+1), 2)
+            + __pad(curDate.getDate(), 2)
+            + separator
+            + __pad(curDate.getHours(), 2)
+            + __pad(curDate.getMinutes(), 2)
+            + __pad(curDate.getSeconds(), 2);
+        return dateFormat + "." + extName;
```
Comment (YoungeunKim) [inline lib/util/createFileName.js:28]: if(extName) {
 return dateFormat + "." + extName;
} else {
 return dateFormat;
}

It seems that line 16~18 can be removed.

File: lib/log.js [hunk 9]
```diff
@@ -339,7 +340,9 @@ const util = require('util'),
     function createLogFile(options, next) {
         log.info("log#createLogFile");
         if (options.argv.argv.remain.length === 0) {
-            options.argv.argv.remain[0] = path.resolve(_createDefaultFileName());
+            const fileNameSeperator = "_",
+                fileExt = "log";
+            options.argv.argv.remain[0] = path.resolve(createDateFileName(fileNameSeperator, fileExt));
         } else {
             if (path.extname(options.argv.argv.remain[0]) === "") {
                 options.argv.argv.remain[0] = options.argv.argv.remain[0] + ".log";
```
Comment (YoungeunKim) [inline lib/log.js:348]: can you use "fileExt" instead  of "log"? (It's optional)

File: lib/device.js [hunk 10]
```diff
@@ -269,28 +257,23 @@ const util = require('util'),
...
-                if (options.outputPath === null){
-                    captureFileName = _createDateFileName();
+                if (options.outputPath === null) {
+                    captureFileName += createDateFileName(null, captureFormat.toLowerCase());
                     destinationPath = path.resolve('.');
                 } else {
                     // Get directory path, file path, file Extenstion from given path
                     const parseDirPath = path.dirname(options.outputPath),
                         parseBase = path.parse(options.outputPath).base;
+                    console.log(parseBase);
```
Comment (YoungeunKim) [inline lib/device.js:276]: Please remove if it is not required.

**Review by YoungeunKim — CHANGES_REQUESTED** [review]

**Review by ssuminahn — COMMENTED** [review]
code :+1: 
verify :+1:

File: lib/device.js (hunk 10 above)
Comment (ssuminahn) [reply lib/device.js]: I've updated.

File: lib/log.js (hunk 9 above)
Comment (ssuminahn) [reply lib/log.js]: I've updated it.

File: lib/util/createFileName.js (hunk 8 above)
Comment (ssuminahn) [reply lib/util/createFileName.js]: I've updated it.

File: lib/launch.js [hunk 11]
```diff
@@ -369,6 +351,17 @@ const util = require('util'),
         }
     };
 
+    function makeSession(options, next){
+        log.info("launch#makeSession");
+        options.nReplies = 1; // -n 1
+        if (!options.session) {
+            const printTarget = true;
+                    options.session = new novacom.Session(options.device, printTarget, next);
```
Comment (YoungeunKim) [inline lib/launch.js:359]: Please remove spaces in front of "options.session" keyword.

**Review by YoungeunKim — CHANGES_REQUESTED** [review]
Verified 👍 

Please fix minor thing.(indentation)

**Review by ssuminahn — COMMENTED** [review]
code :+1: 
verify :+1:

File: lib/launch.js (hunk 11 above)
Comment (ssuminahn) [reply lib/launch.js]: I've update it.

**Review by YoungeunKim — APPROVED** [review]
code : 👍 
verified : 👍

**Pull request merged**